src/main/java/VASSAL/build/module/map/CounterDetailViewer.java

```diff
@@ -652,8 +656,11 @@ protected void apply(GamePiece p) {
           lastLayer = layer;
         }
 
-        if (foundPieceAt == null) {
-          foundPieceAt = p.getPosition();
+        // Our show-overlap setting controls whether we're allowed to show all pieces overlapping this spot, or only pieces at the exact location of the first piece we find.
+        if (!showingOverlap) {
+          if (foundPieceAt == null) {
+            foundPieceAt = p.getPosition();
+          }
```

Comment:
Test the conditions together since the only line which does anything is enclosed in both blocks.

`if (!showingOverlap && foundPieceAt == null) {`

Reply:
Done